### DISCUSSION TITLE
Copilot vim is taking so much RAM and CPU

### DISCUSSION THREAD

**Discussion by JRS-Developer** [community/community#30864] [Copilot Conversations]
Hello, I have a problem with the plugin copilot.vim. This is my neovim version:
NVIM v0.7.2
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3

And I'm using copilot.vim version 1.5.2

When I close neovim, copilot continues open consuming even 110% of my CPU

![image](https://user-images.githubusercontent.com/77207702/186792066-7604fcce-344e-4729-97b0-cfec32a7ae1b.png)

.

**Comment by tpope** [comment]
Can you tell me what Node version you're using? Check `:echo system(['node', '--version'])`

**Reply by JRS-Developer** [reply]
I'm using node 16.17.0

**Reply by tpope** [reply]
Can confirm that version works fine. One thing you could try is invoke `:call copilot#Agent().Close()` and see if that also causes the problem to occur. If so, check `~/.cache/nvim/lsp.log` and see if there are any clues there. You can also do `:lua vim.lsp.set_log_level("TRACE")` to enable more verbose logging.

**Reply by JRS-Developer** [reply]
No, it doesn't, I don't know why but now It's working fine.

**Comment by hamidzr** [comment]
Hello I might be having the same issue here. I'll give these steps a try. It seems to stay on even after all vim processes are shut down.

**Reply by hamidzr** [reply]
hopefully, there isn't something here that shouldn't have been logged. I see a lot of unhandled exception from RPC calls
[lsp.log](https://github.com/community/community/files/9592689/lsp.log)

**Reply by tpope** [reply]
That's a known issue on Node 18.

**Reply by hamidzr** [reply]
Could it happen on 16 as well? I'm on v16.16.0 and don't have 18 installed. lmk if I can provide anything else that might help

**Reply by tpope** [reply]
Not sure, but the issue should be fixed in the latest copilot.vim, which might solve it for you too.

**Reply by hamidzr** [reply]
thanks, I upgraded to `Copilot.vim 1.5.4` I can report what I find in the coming days here if it helps

**Reply by hamidzr** [reply]
FWIW I haven't noticed this anymore.

**Comment by olimorris** [comment]
I ran into this issue recently and mistakenly assumed it was `copilot` that was the culprit. In fact, it was the `tailwindcss` lsp server
